contracts/SpokePool.sol

```diff
@@ -1393,6 +1393,8 @@ abstract contract SpokePool is
         // This method by default is a no-op. Different child spoke pools might want to execute functionality here
         // such as wrapping any native tokens owned by the contract into wrapped tokens before proceeding with
         // executing the leaf.
+
+        if (address(msg.sender).isContract()) revert NotEOA();
```

Comment:
This will prevent us from bundling HubPool and SpokePool executions together, no? That'd be unfortunate because it's actually really nice to have those executed atomically.